test/SourceKit/CursorInfo/cursor_in_pound_if.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %sourcekitd-test -req=cursor -pos=6:7 %s -- %s | %FileCheck %s --check-prefix=CHECK-INT
+// RUN: %sourcekitd-test -req=cursor -pos=8:7 %s -- %s | %FileCheck %s --check-prefix=CHECK-STR
+
+func foo() {
+#if USE_INT
+  let xxx = 1
+#else
+  let xxx = "hello"
+#endif
+}
+// TODO: Once we switch to use the solver-based cursor info implementation, we also receive results for the int case
```

Comment:
Does it handle unparsed `#if` regions?

Reply:
Yes, it just fell out of the implementation that it does. I think we just type check whatever decl the cursor info points to. I’m not entirely sure how well it works if the decl you invoke cursor info on references other variables in a `#if` though.